File: download_unique.js
const fs = require('fs');
const https = require('https');
const path = require('path');

const products = [
    { name: "Phantom Oversized Tee", slug: "phantom-oversized-tee", q: "t-shirt fashion" },
    { name: "Ghost Drop Tee", slug: "ghost-drop-tee", q: "t-shirt fashion" },
    { name: "Midnight Crop Top", slug: "midnight-crop-top", q: "crop top fashion" },
    { name: "Raw Edge Tank", slug: "raw-edge-tank", q: "tank top fashion" },
    { name: "Stealth Hoodie", slug: "stealth-hoodie", q: "hoodie streetwear" },
    { name: "Cloud Nine Pullover", slug: "cloud-nine-pullover", q: "pullover sweatshirt" },
    { name: "Neon Zip Hoodie", slug: "neon-zip-hoodie", q: "zip hoodie" },
    { name: "Crew Neck Sweatshirt", slug: "crew-neck-sweatshirt", q: "crew neck sweatshirt" },
    { name: "Urban Cargo Jogger", slug: "urban-cargo-jogger", q: "cargo pants streetwear" },
    { name: "Tech Fleece Pant", slug: "tech-fleece-pant", q: "fleece pants fashion" },
    { name: "Slim Track Pant", slug: "slim-track-pant", q: "track pants" },
    { name: "Wide Leg Cargo", slug: "wide-leg-cargo", q: "wide leg cargo pants" },
    { name: "Drift Training Shorts", slug: "drift-training-shorts", q: "training shorts" },
    { name: "Mesh Basketball Short", slug: "mesh-basketball-short", q: "basketball shorts" },
    { name: "Compression Liner Short", slug: "compression-liner-short", q: "compression shorts" },
    { name: "Cargo Utility Short", slug: "cargo-utility-short", q: "cargo shorts" },
    { name: "Shadow Windbreaker", slug: "shadow-windbreaker", q: "windbreaker jacket" },
    { name: "Puffer Vest", slug: "puffer-vest", q: "puffer vest" },
    { name: "Coach Jacket", slug: "coach-jacket", q: "coach jacket" },
    { name: "Track Jacket", slug: "track-jacket", q: "track jacket" },
    { name: "Snapback Cap", slug: "snapback-cap", q: "snapback cap" },
    { name: "Bucket Hat", slug: "bucket-hat", q: "bucket hat fashion" },
    { name: "Crossbody Bag", slug: "crossbody-bag", q: "crossbody bag" },
    { name: "Crew Socks (3-Pack)", slug: "crew-socks-3-pack", q: "crew socks" },
    { name: "Duffle Bag", slug: "duffle-bag", q: "duffle bag" },
    { name: "Beanie", slug: "beanie", q: "beanie hat" },
    { name: "Performance Gloves", slug: "performance-gloves", q: "winter gloves" },
    { name: "Logo Belt", slug: "logo-belt", q: "fashion belt" },
    { name: "Oversized Graphic Tee", slug: "oversized-graphic-tee", q: "graphic tee streetwear" },
    { name: "Ribbed Long Sleeve", slug: "ribbed-long-sleeve", q: "long sleeve shirt" },
    { name: "Washed Henley", slug: "washed-henley", q: "henley shirt" },
    { name: "Zip Cargo Pant", slug: "zip-cargo-pant", q: "cargo pants black" }
];

const collections = [
    { slug: "tees", q: "t-shirts rack" },
    { slug: "hoodies", q: "hoodies streetwear" },
    { slug: "joggers", q: "pants streetwear" },
    { slug: "accessories", q: "fashion accessories" }
];

const outputDir = path.join(__dirname, 'public', 'images', 'products');
const collectionsDir = path.join(__dirname, 'public', 'images', 'collections');
const rootImagesDir = path.join(__dirname, 'public', 'images');

[outputDir, collectionsDir, rootImagesDir].forEach(d => {
  if (!fs.existsSync(d)) fs.mkdirSync(d, { recursive: true });
});

function fetchUnsplashURL(query) {
  return new Promise((resolve, reject) => {
    https.get(`https://unsplash.com/napi/search/photos?query=${encodeURIComponent(query)}&per_page=1&page=1`, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        try {
          const json = JSON.parse(data);
          if (json.results && json.results.length > 0) {
            resolve(json.results[0].urls.regular);
          } else {
            resolve("https://images.unsplash.com/photo-1521572163474-6864f9cf17ab?w=800&q=80"); // fallback
          }
        } catch (e) {
          reject(e);
        }
      });
    }).on('error', reject);
  });
}

function download(url, dest) {
  return new Promise((resolve, reject) => {
    const file = fs.createWriteStream(dest);
    https.get(url, (response) => {
      if (response.statusCode >= 300 && response.statusCode < 400 && response.headers.location) {
        return download(response.headers.location, dest).then(resolve).catch(reject);
      }
      response.pipe(file);
      file.on('finish', () => file.close(resolve));
    }).on('error', (err) => fs.unlink(dest, () => reject(err)));
  });
}

async function run() {
  console.log("Downloading unique images...");
  for (const p of products) {
    const dest = path.join(outputDir, `${p.slug}.jpg`);
    console.log(`Fetching ${p.name}...`);
    try {
      const url = await fetchUnsplashURL(p.q);
      await download(url, dest);
    } catch (e) {
      console.error(`Error on ${p.slug}`, e.message);
    }
  }

  console.log("Downloading collection images...");
  for (const c of collections) {
    const dest = path.join(collectionsDir, `${c.slug}.jpg`);
    console.log(`Fetching collection ${c.slug}...`);
    try {
      const url = await fetchUnsplashURL(c.q);
      await download(url, dest);
    } catch (e) {
      console.error(`Error on ${c.slug}`, e.message);
    }
  }

  console.log("Downloading Hero banner...");
  const heroDest = path.join(rootImagesDir, "hero.jpg");
  const heroUrl = await fetchUnsplashURL("streetwear model banner landscape");
  await download(heroUrl, heroDest);

  console.log("Finished generating unique images!");
}

run();

File: download_unique.py
import requests
import json
import os

products = [
    "phantom-oversized-tee", "ghost-drop-tee", "midnight-crop-top", "raw-edge-tank",
    "stealth-hoodie", "cloud-nine-pullover", "neon-zip-hoodie", "crew-neck-sweatshirt",
    "urban-cargo-jogger", "tech-fleece-pant", "slim-track-pant", "wide-leg-cargo",
    "drift-training-shorts", "mesh-basketball-short", "compression-liner-short", "cargo-utility-short",
    "shadow-windbreaker", "puffer-vest", "coach-jacket", "track-jacket",
    "snapback-cap", "bucket-hat", "crossbody-bag", "crew-socks-3-pack",
    "duffle-bag", "beanie", "performance-gloves", "logo-belt",
    "oversized-graphic-tee", "ribbed-long-sleeve", "washed-henley", "zip-cargo-pant"
]

urls = [
    "https://i.imgur.com/QkIa5tT.jpeg", "https://i.imgur.com/jb5Yu0h.jpeg", "https://i.imgur.com/UlxxXyG.jpeg",
    "https://i.imgur.com/1twoaDy.jpeg", "https://i.imgur.com/FDwQgLy.jpeg", "https://i.imgur.com/kg1ZhhH.jpeg",
    "https://i.imgur.com/cHddUCu.jpeg", "https://i.imgur.com/CFOjAgK.jpeg", "https://i.imgur.com/wbIMMme.jpeg",
    "https://i.imgur.com/R2PN9Wq.jpeg", "https://i.imgur.com/IvxMPFr.jpeg", "https://i.imgur.com/7eW9nXP.jpeg",
    "https://i.imgur.com/cSytoSD.jpeg", "https://i.imgur.com/WwKucXb.jpeg", "https://i.imgur.com/cE2Dxh9.jpeg",
    "https://i.imgur.com/ZKGofuB.jpeg", "https://i.imgur.com/GJi73H0.jpeg", "https://i.imgur.com/633Fqrz.jpeg",
    "https://i.imgur.com/mp3rUty.jpeg", "https://i.imgur.com/JQRGIc2.jpeg", "https://i.imgur.com/9LFjwpI.jpeg",
    "https://i.imgur.com/vzrTgUR.jpeg", "https://i.imgur.com/p5NdI6n.jpeg", "https://i.imgur.com/R3iobJA.jpeg",
    "https://i.imgur.com/Wv2KTsf.jpeg", "https://i.imgur.com/76HAxcA.jpeg", "https://i.imgur.com/wXuQ7bm.jpeg",
    "https://i.imgur.com/BZrIEmb.jpeg", "https://i.imgur.com/KcT6BE0.jpeg", "https://i.imgur.com/cBuLvBi.jpeg",
    "https://i.imgur.com/N1GkCIR.jpeg", "https://i.imgur.com/kKc9A5p.jpeg"
]

output_dir = "public/images/products"
os.makedirs(output_dir, exist_ok=True)

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8",
    "Referer": "https://imgur.com/"
}

print(f"Downloading {len(products)} images...")

for i, product in enumerate(products):
    url = urls[i]
    dest = os.path.join(output_dir, f"{product}.jpg")
    print(f"Downloading {product}...")
    try:
        img_res = requests.get(url, headers=headers, timeout=10)
        if img_res.status_code == 200:
            with open(dest, 'wb') as f:
                f.write(img_res.content)
            print(f"  Saved {len(img_res.content)} bytes.")
        else:
            print(f"  Failed status: {img_res.status_code}")
    except Exception as e:
        print(f"Failed to download {product}: {e}")
        
print("Done downloading unique images!")
